packages/langium/src/workspace/workspace-lock.ts

```diff
@@ -59,8 +72,26 @@ export class DefaultWorkspaceLock implements WorkspaceLock {
         return this.enqueue(this.writeQueue, action, tokenSource.token);
     }
 
-    read<T>(action: () => MaybePromise<T>): Promise<T> {
-        return this.enqueue(this.readQueue, action);
+    read<T>(action: () => MaybePromise<T>, priority?: WorkspaceLockPriority): Promise<T> {
+        if (priority === WorkspaceLockPriority.Immediate) {
+            this.counter++;
+            const deferred = new Deferred<T>();
+            (async () => {
+                try {
+                    await delayNextTick();
+                    const result = await action();
+                    deferred.resolve(result);
+                } catch (err) {
+                    deferred.reject(err);
+                } finally {
+                    this.counter--;
+                    this.performNextOperation();
+                }
+            })();
+            return deferred.promise;
```

Comment:
I think you can get rid of the `Deferred` if you `return` the result in the inline function and either `return` the inline function's result or eliminate it by making the whole method `async`.